src/GraphBLAS-sharp.Backend/Vector/DenseVector/DenseVector.fs

```diff
@@ -0,0 +1,317 @@
+namespace GraphBLAS.FSharp.Backend
+
+open Brahma.FSharp
+open GraphBLAS.FSharp.Backend
+open GraphBLAS.FSharp.Backend.Common
+open Microsoft.FSharp.Quotations
+
+module DenseVector =
+    let private fillMask<'a, 'b when 'a: struct and 'b: struct> (clContext: ClContext) (workGroupSize: int) =
+
+        let fillMask =
+            <@ fun (ndRange: Range1D) length (maskArray: ClArray<'a option>) (scalar: ClCell<'b>) (resultArray: ClArray<'b option>) ->
+
+                let gid = ndRange.GlobalID0
+
+                if gid < length then
+                    match maskArray.[gid] with
+                    | Some _ -> resultArray.[gid] <- Some scalar.Value
+                    | None -> resultArray.[gid] <- None @>
+
+        let kernel = clContext.Compile(fillMask)
+
+        fun (processor: MailboxProcessor<_>) (maskVector: ClArray<'a option>) (scalarCell: ClCell<'b>) ->
+
+            let resultArray =
+                clContext.CreateClArray(
+                    maskVector.Length,
+                    hostAccessMode = HostAccessMode.NotAccessible,
+                    deviceAccessMode = DeviceAccessMode.ReadWrite,
+                    allocationMode = AllocationMode.Default
+                )
+
+            let ndRange =
+                Range1D.CreateValid(maskVector.Length, workGroupSize)
+
+            let kernel = kernel.GetKernel()
+
+            processor.Post(
+                Msg.MsgSetArguments
+                    (fun () -> kernel.KernelFunc ndRange maskVector.Length maskVector scalarCell resultArray)
+            )
+
+            processor.Post(Msg.CreateRunMsg<_, _>(kernel))
+
+            resultArray
+
+    let elementWiseAddAtLeasOne<'a, 'b, 'c when 'a: struct and 'b: struct and 'c: struct>
```

Comment:
Since it's dense vector, should we use AtLeastOne? @gsvgit

Reply:
I guess we should provide this function for a unified interface for different representations of vectors, at least. Again, Vector (in GraphBLAS) explicitly contains zeroes as None, so it is natural to provide function operates over atLeastOne.